source/server/listener_manager_impl.cc

```diff
@@ -88,19 +90,47 @@ ListenerImpl::ListenerImpl(const envoy::api::v2::Listener& config, ListenerManag
       local_drain_manager_(parent.factory_.createDrainManager()) {
   // TODO(htuch): Support multiple filter chains #1280, add constraint to ensure we have at least on
   // filter chain #1308.
-  ASSERT(config.filter_chains().size() == 1);
-  const auto& filter_chain = config.filter_chains()[0];
-
-  listener_scope_ =
-      parent_.server_.stats().createScope(fmt::format("listener.{}.", address_->asString()));
+  ASSERT(config.filter_chains().size() >= 1);
+
+  // Skip lookup and update of the SSL Context if there is only one filter chain
+  // and it doesn't enforce any SNI restrictions.
+  bool skip_context_update;
+  if (config.filter_chains().size() == 1 &&
+      config.filter_chains()[0].filter_chain_match().sni_domains().empty()) {
+    skip_context_update = true;
+  } else {
+    skip_context_update = false;
```

Comment:
nit: initialize to false on line 97 and remove the else clause.

Reply:
Better yet: skip_context_update = (config.filter_chains().size() == 1 && config.filter_chains()[0].filter_chain_match().sni_domains().empty());

Reply:
Done.